impeller/display_list/display_list_dispatcher.cc

```diff
@@ -298,7 +340,20 @@ void DisplayListDispatcher::setColorSource(
       paint_.contents = std::move(contents);
       return;
     }
-    case flutter::DlColorSourceType::kImage:
+    case flutter::DlColorSourceType::kImage: {
+      const flutter::DlImageColorSource* image_color_source = source->asImage();
+      FML_DCHECK(image_color_source &&
+                 image_color_source->image()->impeller_texture());
+      auto contents = std::make_shared<TiledTextureContents>();
+      contents->SetTexture(image_color_source->image()->impeller_texture());
+      contents->SetTileModes(
+          ToTileMode(image_color_source->horizontal_tile_mode()),
+          ToTileMode(image_color_source->vertical_tile_mode()));
+      contents->SetSamplerDescriptor(
+          ToSamplerDescriptor(image_color_source->sampling()));
+      paint_.contents = std::move(contents);
```

Comment:
No need to implement it now, but mind dropping a TODO nearby for handling the `source->asImage()->matrix()` parameter as described [here](https://api.flutter.dev/flutter/dart-ui/ImageShader-class.html). In this case, I think all of the texture coordinates should just be multiplied with the matrix.

I think we need to support it for all of the other color sources as well.

Reply:
Issue link added